### PR TITLE
Views Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
### In this project I added the Views associated with the Models and Controllers I already created for the Blog App. The following are the changes:

- Implemented the design from the sneak peek wireframes [1](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_root_users.png), [2](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_page.png), [3](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png), [4](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png).
- Used methods that I have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,28 @@
-<h1>Here is the list of users with a given ID</h1>
\ No newline at end of file
+<div class="user-detail">
+    <div class="detail-description">
+    <div><img src="<%= @user.photo %>" alt="User" /></div>
+    <div class="user-desc">
+        <h1> <%= @user.name %></h1>
+        <h3>Number of posts: <%= @user.posts.count %></h3>
+    </div>
```
Comment (FrederickMih) [inline app/views/users/show.html.erb]: - [Optional] I think the reason for using partials is to keep our codebase DRY by avoiding repeating the same piece of code. You got a partial for this piece of code, I think you were just supposed to render it when needed.

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<a href="/users/<%= user.id %>">
+  <div class="users-list">
+    <img src="<%= user.photo %>" alt="User" />
```
Comment (FrederickMih) [inline app/views/users/_user.html.erb]: - [ ] Please fix your design. The image is extremely large and it renders the final design different from the wireframe. Image comes first if you look at the wireframe critically

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,36 @@
...
+                </p>
+                <div class="comment-like">
+                    <span>Comments: <%= post.comment_counter%> </span>
+                    <span class="likes">Likes:<%= post.likes_counter%></span>
+                </div>
+
+                <div class="comments-container">
+                    <% post.recent_five_comments.each do |comment| %>
+                        <div  id="single-comment"> 
+                            <span><%= comment.text %></span>
```
Comment (FrederickMih) [inline app/views/posts/index.html.erb:28]: - [ ] I think you're missing the username of the commenter as seen in the original design. I'll advise you to display the username to meet up with the requirements.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,36 @@
...
+                    <h1><%= @user.name %></h1>
+                    <h3>Number of posts: <%= @posts.count %>
+                    </h3>
+                </div>
+        </div>
+
+    <div class="recent-posts">
+        <% @posts.each do |post| %>
+        <div class="one-comment">
+                <h2><%= post.title %>
```
Comment (FrederickMih) [inline app/views/posts/index.html.erb:15]: - [ ] To enhance the user's experience let's make sure clicking on each post, takes the user to the post show page.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @HERMON-1995, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (HERMON-1995) [reply app/views/posts/index.html.erb:15]: Thank you! Clicking on each post, takes the user to the post show page. 😊

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (HERMON-1995) [reply app/views/posts/index.html.erb:28]: Sir! Can you please check it in this file [line](https://github.com/HERMON-1995/Blog_app_project/blob/639e0cc70ea77edf4444dac4f1e0493605d4aa4a/app/views/posts/show.html.erb#L19), Even in the browser it is rendering fine with the author's name and comment side by side.

File: app/views/users/_user.html.erb (hunk 2 above)
Comment (HERMON-1995) [reply app/views/users/_user.html.erb]: I implemented the style, now the images are smaller, thank you!

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (Qoosim) [inline app/views/posts/index.html.erb:15]: - [ ] As suggested by the previous reviewer, your design must match the wireframe provided. The image of the user at this moment is extremely large, and as a result distorted the page layout in a way. Kindly look at the wireframe below and make your layout to look like it. Styling is not required at this moment though, but the wireframe layout must be maintained.

### Your Design
user-img1

user-img2

### Wireframe
wireframe

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @HERMON-1995, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (HERMON-1995) [reply app/views/posts/index.html.erb:15]: Ohh sorry about that, last time I only implemented the image section on the home page. now users post image is also fixed, Thank you! 😁

**Review by DammyShittu — APPROVED** [review]
Hi @HERMON-1995,

## APPROVED 🟢

## Highlights
✔️ Good layout

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HERMON-1995** [comment]
Thanks for approving my PR! 😄

**Pull request merged**